### DISCUSSION TITLE
Multi-tenant metadata database

### DISCUSSION THREAD

**Discussion by GoldenMouse** [apache/superset#14299] [Q&A / Help]
Compliance requires we keep our client's data separate.  
How do we do this for the metadata database?  Can we have multiple metadata databases for a given Superset instance?
In general, for a Superset cluster, how we do design a multitenant setup?

**Comment by jgomezb** [comment]
Not an expert, just used the tool and navigated the code.
I think currently it is not possible to have a single Superset instance keeping multiple metadata databases for different clients. The way I would approach this problem would be with a Superset instance per client, where the metadata database is specific in each one. This would require having some templating/argument passing logic for the superset_config.py at deployment time.
